include/Server.hpp

```diff
@@ -36,7 +37,7 @@ class Server
 		Server(char *port);
 		~Server(void);
 
-		void		listen(void);
+		void		listen(Configuration &config);
```

Comment:
The configuration should be a reference in the constructor. This could replace the port as it is stored inside the configuration